pulsar-websocket/src/main/java/org/apache/pulsar/websocket/ProducerHandler.java

```diff
@@ -315,30 +351,117 @@ protected ProducerBuilder<byte[]> getProducerBuilder(PulsarClient client) {
+            } else {
+                keyMetadataArray[index] = entry.getValue().getMetadata().entrySet().stream()
+                        .map(e -> new KeyValue().setKey(e.getKey()).setValue(e.getValue()))
+                        .collect(Collectors.toList());
+            }
+            builder.addEncryptionKey(keyNameArray[index]);
         }
+        // Disable server-side batch process, compression, and encryption, and only set the message metadata that
+        // which client-side provided.
+        builder.enableBatching(false);
+        builder.compressionType(CompressionType.NONE);
+        builder.cryptoKeyReader(DummyCryptoKeyReaderImpl.INSTANCE);
+        // Disable chunking.
+        builder.enableChunking(false);
+        // Inject encryption metadata decorator.
+        builder.messageCrypto(new WSSDummyMessageCryptoImpl(msgMetadata -> {
+            for (int i = 0; i < keyNameArray.length; i++) {
+                msgMetadata.addEncryptionKey().setKey(keyNameArray[i]).setValue(keyValueArray[i])
+                        .addAllMetadatas(keyMetadataArray[i]);
+            }
+        }));
+        // Do warning param check and print warning log.
+        printWarnLogIfSettingBatchedParams();
+        printWarnLogIfSettingCompressionParams();
+    }
 
-        if (queryParams.containsKey("messageRoutingMode")) {
-            checkArgument(
-                    Enums.getIfPresent(MessageRoutingMode.class, queryParams.get("messageRoutingMode")).isPresent(),
-                    "Invalid messageRoutingMode %s", queryParams.get("messageRoutingMode"));
-            MessageRoutingMode routingMode = MessageRoutingMode.valueOf(queryParams.get("messageRoutingMode"));
-            if (!MessageRoutingMode.CustomPartition.equals(routingMode)) {
-                builder.messageRoutingMode(routingMode);
+    private void popularProducerBuilderForServerSideEncrypt(ProducerBuilder<byte[]> builder) {
+        this.clientSideEncrypt = false;
+        if (queryParams.containsKey("batchingEnabled")) {
+            boolean batchingEnabled = Boolean.parseBoolean(queryParams.get("batchingEnabled"));
+            if (batchingEnabled) {
+                builder.enableBatching(true);
+                if (queryParams.containsKey("batchingMaxMessages")) {
+                    builder.batchingMaxMessages(Integer.parseInt(queryParams.get("batchingMaxMessages")));
+                }
+
+                if (queryParams.containsKey("maxPendingMessages")) {
+                    builder.maxPendingMessages(Integer.parseInt(queryParams.get("maxPendingMessages")));
+                }
+
+                if (queryParams.containsKey("batchingMaxPublishDelay")) {
+                    builder.batchingMaxPublishDelay(Integer.parseInt(queryParams.get("batchingMaxPublishDelay")),
+                            TimeUnit.MILLISECONDS);
+                }
+            } else {
+                builder.enableBatching(false);
+                printWarnLogIfSettingBatchedParams();
```

Comment:
Should be removed?

Reply:
No, it is the original design: Web socket proxy will composite multi messages into a batched message if users set the param "batchingEnabled" as `true`. I just moved this code into the method `popularProducerBuilderForServerSideEncrypt()`